client/tailscale/tailscale.go

```diff
@@ -65,20 +65,28 @@ func (c *Client) httpClient() *http.Client {
 	return http.DefaultClient
 }
 
+// DoNotEscape is a URL path element passed to BuildURL and BuildTailnetURL
+// that should not be escaped. Do not use this type for any user-provided
+// input.
+type DoNotEscape string
```

Comment:
Can we do without this type? And just make BuildURL handle things containing `?` specially?

Reply:
Or! Can you instead say that the last argument can be of type `url.Values` and then you build it appropriately, adding the question mark if non-empty and appending https://pkg.go.dev/net/url#Values.Encode ?

Reply:
I like that too.

Reply:
The other option is to have BuildURL return an actual `url.URL`. The `tailscale-client-go-v2` [equivalent](https://github.com/tailscale/tailscale-client-go-v2/blob/main/client.go#L196) of this function (which I poorly translated) does that.

Reply:
Given how the client is structured in this repo, plumbing through the error from trying to parse the string BaseURL into a `url.URL` is probably too much of a hassle, so special handling for `url.Values` seems like a better option.

Reply:
Good idea, done